src/test/utils/base.t.sol

```diff
@@ -82,6 +84,7 @@ contract HookProtocolTest is Test, EIP712, PermissionConstants {
   }
 
   function setUpFullProtocol() public {
+    exec = new Exec();
     weth = new WETH();
     protocol = new HookProtocol(
```

Comment:
does it make sense to have `exec` live on `HookProtocol` and then just make the `HookProtocol` address that all of these contracts already store the `exec` address?